RUFAS/biophysical/manure/storage/anaerobic_lagoon.py

```diff
@@ -0,0 +1,242 @@
+
+        if not manure_to_return:
+            self._stored_manure = copy(self._manure_to_process)
+
+        self._report_manure_stream(self._manure_to_process, "accumulated", time.simulation_day)
+        self._report_manure_stream(received_manure, "received", time.simulation_day)
+
+        function_name = self.process_manure.__name__
+        self._report_processor_output(
+            "methane", total_storage_methane, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
+        )
+        self._report_processor_output(
+            "methane_burned", storage_methane_burned, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
+        )
+        self._report_processor_output(
+            "ammonia_N", storage_ammonia, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
+        )
+        self._report_processor_output(
+            "nitrous_oxide_N", nitrous_oxide_emissions, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
+        )
+
+        return manure_to_return
+
+    def _apply_methane_emissions(self, manure_temperature: float) -> tuple[float, float]:
+        """
+        This method computes the methane emissions from both degradable and non-degradable volatile solids in the
+        manure, adjusts the manure's composition based on the amount of methane emitted, and accounts for the burning
+        of methane if the storage cover is a cover and flare system.
+
+        It applies the methane emissions to self._manure_to_process in-place.
+
+        Parameters
+        ----------
+        manure_temperature : float
+            The temperature of the manure in, (degrees Celsius).
+
+        Returns
+        -------
+        tuple[float, float]
+            A tuple containing:
+            - The methane burned from manure storage on the current day, (kg).
+            - The methane emitted from manure storage on the current day, (kg).
+
+        """
+        storage_methane_from_degradable_volatile_solids = self._calculate_methane_emissions(
+            volatile_solids=self._manure_to_process.degradable_volatile_solids,
+            manure_temperature=manure_temperature,
+            is_degradable=True,
+        )
+        storage_methane_from_non_degradable_volatile_solids = self._calculate_methane_emissions(
+            volatile_solids=self._manure_to_process.non_degradable_volatile_solids,
+            manure_temperature=manure_temperature,
+            is_degradable=False,
+        )
+        total_methane = (
+            storage_methane_from_degradable_volatile_solids + storage_methane_from_non_degradable_volatile_solids
+        )
+        storage_methane_burned = 0.0
+        if self._cover == StorageCover.COVER_AND_FLARE:
+            storage_methane_burned, adjusted = self._calculate_cover_and_flare_methane(total_methane)
+            total_methane = adjusted
+
+        carbon_loss = total_methane * METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO
```

Comment:
```suggestion
        mass_loss = total_methane * METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO
```
This is just me being nitpicky, but I just want to modify this to avoid confusion because this value is not just carbon loss - the H and O (and probably other elements in reality) that are lost when CO2 and CH4 are produced in anaerobic fermentation are also included 💩